qiskit/result/result.py

```diff
@@ -181,6 +182,87 @@ def get_counts(self, experiment=None):
         except KeyError:
             raise QiskitError('No counts for experiment "{0}"'.format(experiment))
 
+    def get_assertion(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+
+        Returns:
+            tuple: tuple containing:
+
+                chisq(float): the chi-squared value
+                pval(float): the p-value
+                passed(Boolean): if assertion passed
+        """
+        exp = experiment
+        counts = self.get_counts(exp)
+        assertion = exp.data[-1][0]
+
+        cbits = assertion._cbit
+        cbits = Asserts.clbits2idxs(cbits, exp)
+
+        new_counts = {}
+        for (key, value) in counts.items():
+            new_key = key.replace(' ', '')
+            new_key = ''.join([new_key[-1*(cbit+1)] for cbit in cbits][::-1])
+            new_counts.setdefault(new_key, 0)
+            new_counts[new_key] += value
+        counts = new_counts
```

Comment:
it seems unnecessary

Reply:
This enables assertions to be performed on parts of a result.
The counts dictionary of a result is for all cbits in the quantum circuit.
Say you wanted to make an assertion on just some of the qubits in the quantum circuit, and not all of them, this piece of code builds a new counts dictionary just for the cbits the assertion is on.
Thus, for a state that is |0>|+>|1>, you can assertclassical that the 1st and 3rd qubits are "01", and assertsuperposition on the 2nd qubit.

Reply:
What I mean is `chisq, pval, passed = assertion.stat_test(new_counts)`....

Reply:
Also, consider to remove the `get_` in the methods names https://github.com/Qiskit/qiskit-terra/issues/1346

Reply:
This is fixed by 954105393d451637bf3076c066dd86afccbe5000